gprofiler/profilers/java.py

```diff
@@ -730,8 +829,13 @@ def __init__(
         self._pid_to_java_version: Dict[int, Optional[str]] = {}
         self._kernel_messages_provider = get_kernel_messages_provider()
         self._enabled_proc_events_java = False
+        self._java_jvm_flags_to_retrieve = self._init_java_flag_retrieval(java_jvm_flags_to_retrieve)
+        self._jattach_jcmd_runner = JattachJcmdRunner(
+            stop_event=self._stop_event, jattach_timeout=self._jattach_timeout
+        )
         self._ap_timeout = self._duration + self._AP_EXTRA_TIMEOUT_S
-        self._metadata = JavaMetadata(self._stop_event)
+        application_identifiers.ApplicationIdentifiers.init_java(self._jattach_jcmd_runner)
```

Comment:
This has the (hidden?) assumption that java appids are used only from the `JavaProfiler`, i.e if the `JavaProfiler` was not initialized, it's okay that we don't have java appids. I checked now and it seems fine, and I guess, in any case we will get a `KeyError`...

Reply:
yes we will, it was the same with `set_enrichment_options` wasn't it?

Reply:
That wasn't called in `JavaProfiler` though, but in some generic code.

Anyway, we will get an explicit error, so I'm fine with it